pkg/scraper/jobcenter_scraper.go

```diff
@@ -4,67 +4,89 @@ import (
 	"errors"
 	"fmt"
 	"regexp"
+	"strconv"
+	"strings"
+	"sync"
 
 	"github.com/PuerkitoBio/goquery"
 	"github.com/b-open/jobbuzz/pkg/model"
 )
 
 const (
-	pageSize = 30
+	pageSize = 200
 
 	jobcenterUrl = "https://www.jobcentrebrunei.gov.bn"
 )
 
+var jobcenterJobs = []*model.Job{}
+
 func ScrapeJobcenter() ([]*model.Job, error) {
+	var wg sync.WaitGroup
 
-	jobs := []*model.Job{}
+	lastPageNo, err := scrapeJobcenterLastPageNumber()
+	if err != nil {
+		return nil, err
+	}
 
-	url := fmt.Sprintf("%s/web/guest/search-job?q=&delta=%d", jobcenterUrl, pageSize)
+	for i := 1; i <= lastPageNo; i++ {
+		wg.Add(1)
+		url := fmt.Sprintf("%s/web/guest/search-job?q=&delta=%d&start=%d", jobcenterUrl, pageSize, i)
+		go scrapeJobcenterJobsListing(url, &wg)
+	}
+
+	wg.Wait()
+
+	return jobcenterJobs, nil
+}
+
+func scrapeJobcenterJobsListing(url string, wg *sync.WaitGroup) {
+	defer wg.Done()
 
 	doc, err := getDocument(url)
 	if err != nil {
-		return nil, err
+		fmt.Printf("Fail to scrape url : %s, err: %s \n", url, err)
 	}
 
-	doc.Find("li.list-group-item.list-group-item-flex").EachWithBreak(func(i int, s *goquery.Selection) bool {
-
-		jobTitle := s.Find(".jp_job_post_right_cont h4 a").Text()
-		company := s.Find(".jp_job_post_right_cont p a").Text()
-		salary := s.Find(".jp_job_post_right_cont>ul li:first-child").Text()
-		location := s.Find(".jp_job_post_right_cont>ul li:nth-child(2)").Text()
-
-		link, exist := s.Find(".jp_job_post_right_cont h4 a").Attr("href")
-		if !exist {
-			return true
-		}
-
-		providerJobId, err := getJobcenterJobId(link)
-		if err != nil {
-			return true
-		}
-
-		description, err := scrapeJobDescription(link)
-		if err != nil {
-			return true
-		}
-
-		job := model.Job{
-			Provider:      JobCenter,
-			ProviderJobId: providerJobId,
-			Title:         jobTitle,
-			Company:       company,
-			Salary:        salary,
-			Location:      location,
-			Link:          link,
-			Description:   *description,
-		}
-
-		jobs = append(jobs, &job)
-
-		return true
+	doc.Find("li.list-group-item.list-group-item-flex").Each(func(i int, s *goquery.Selection) {
+		wg.Add(1)
+
+		go func(s *goquery.Selection) {
+			defer wg.Done()
+			jobTitle := s.Find(".jp_job_post_right_cont h4 a").Text()
+			company := s.Find(".jp_job_post_right_cont p a").Text()
+			salary := s.Find(".jp_job_post_right_cont>ul li:first-child").Text()
+			location := s.Find(".jp_job_post_right_cont>ul li:nth-child(2)").Text()
+
+			link, exist := s.Find(".jp_job_post_right_cont h4 a").Attr("href")
+			if !exist {
+				fmt.Printf("Fail to scrape job link : %s, err: %s \n", link, err)
+			}
+
+			providerJobId, err := getJobcenterJobId(link)
+			if err != nil {
+				fmt.Printf("Fail to scrape job id for link : %s, err: %s \n", link, err)
```

Comment:
Does this not need a return?
Is it ok for this code to continue with the rest of the code when the id is missing?
Same for line 72